### PR TITLE
[c2f,bug,#33][s]: Add test against empty `keywords`

### PR DESCRIPTION
**Pull request by sglavoie**
Fix bug discussed in #33 on generating `'keywords': []` when we have `'tags': []` from converting from CKAN to Frictionless.

* Add test_ckan_round_trip_does_not_generate_empty_keywords.
* Do not create empty list of `keywords` when tags are empty.

### CODE REVIEW

File: frictionless_ckan_mapper/ckan_to_frictionless.py [hunk 1]
```diff
@@ -111,6 +111,8 @@ def dataset(ckandict):
     if 'tags' in ckandict:
         outdict['keywords'] = [tag['name'] for tag in ckandict['tags']]
         del outdict['tags']
+    if outdict.get("keywords") == []:
```
Comment (rufuspollock) [inline frictionless_ckan_mapper/ckan_to_frictionless.py:114]: more pythonic is just:

```
if not outdict.get("keywords"):
```

as python evaluates `[]` to False.

File: frictionless_ckan_mapper/ckan_to_frictionless.py [hunk 2]
```diff
@@ -111,6 +111,8 @@ def dataset(ckandict):
...
+        del outdict["keywords"]
```
Comment (rufuspollock) [inline frictionless_ckan_mapper/ckan_to_frictionless.py:115]: won't this error is `keywords` is not in the dict ... (or is it always in the dict?)

File: frictionless_ckan_mapper/ckan_to_frictionless.py [hunk 3]
```diff
@@ -111,6 +111,8 @@ def dataset(ckandict):
     if 'tags' in ckandict:
```
Comment (rufuspollock) [inline frictionless_ckan_mapper/ckan_to_frictionless.py:111]: the better way to go here would be simply:

```
if ckandict.get('tags'):
    ...
```

Since that will only create keywords if tags in ckandict and it is not empty array.

File: tests/test_roundtrip.py [hunk 4]
```diff
@@ -59,3 +59,18 @@ def test_differences_ckan_round_trip(self):
         # - Keys defined in CKAN but ignored in Frictionless, such as `id`
```
Comment (rufuspollock) [inline tests/test_roundtrip.py:59]: We should have a test for this in main unit tests not just roundtrip.

File: frictionless_ckan_mapper/ckan_to_frictionless.py (hunk 2 above)
Comment (sglavoie) [reply frictionless_ckan_mapper/ckan_to_frictionless.py:115]: No, this is why I did it this way:

```
>>> outdict = {}
>>> if not outdict.get("keywords"):
...     del outdict["keywords"]
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
KeyError: 'keywords'

>>> if outdict.get("keywords") == []:
...     del outdict["keywords"]
...
>>>
```

**Comment by rufuspollock** [comment]
Fixed separately.
